ipa-core/src/query/processor.rs

```diff
@@ -349,17 +361,60 @@ impl Processor {
         Some(status)
     }
 
-    /// Returns the query status.
+    /// Returns the query status in this helper, by querying all shards.
     ///
     /// ## Errors
     /// If query is not registered on this helper.
     ///
     /// ## Panics
     /// If the query collection mutex is poisoned.
-    pub fn query_status(&self, query_id: QueryId) -> Result<QueryStatus, QueryStatusError> {
+    pub async fn query_status(
+        &self,
+        shard_transport: ShardTransportImpl,
+        query_id: QueryId,
+    ) -> Result<QueryStatus, QueryStatusError> {
+        let shard_index = shard_transport.identity();
+        if shard_index != ShardIndex::FIRST {
+            return Err(QueryStatusError::NotLeader(shard_index));
+        }
+
         let status = self
             .get_status(query_id)
             .ok_or(QueryStatusError::NoSuchQuery(query_id))?;
+
+        let shard_query_status_req = CompareStatusRequest { query_id, status };
+
+        shard_transport.broadcast(shard_query_status_req).await?;
+
+        Ok(status)
+    }
+
+    /// Compares this shard status against the given type. Returns an error if different.
+    ///
+    /// ## Errors
+    /// If query is not registered on this helper or
+    ///
+    /// ## Panics
+    /// If the query collection mutex is poisoned.
+    pub fn shard_status(
+        &self,
+        shard_transport: &ShardTransportImpl,
+        req: &CompareStatusRequest,
+    ) -> Result<QueryStatus, QueryStatusError> {
+        let shard_index = shard_transport.identity();
+        if shard_index == ShardIndex::FIRST {
+            return Err(QueryStatusError::Leader);
+        }
+        let status = self
```

Comment:
is Codecov hallucinating again or we don't have coverage here?

Reply:
I think we do have tests for this...

Reply:
is it possible to check that by adding a panic and observing that one or more tests fail?